client/state/happychat/middleware.js

```diff
@@ -129,10 +132,19 @@ export const connectChat = ( connection, { getState, dispatch } ) => {
 
 			return sign( { user, session_id } );
 		} )
-		.then( ( { jwt } ) => connection.open( user.ID, jwt, locale ) )
+		.then( ( { jwt } ) => connection.open( user.ID, jwt, locale, groups ) )
 		.catch( e => debug( 'failed to start happychat session', e, e.stack ) );
 };
 
+
+export const updateChatPreferences = ( connection, { getState }, siteId ) => {
+	const state = getState();
+	const locale = getCurrentUserLocale( state );
+	const groups = getGroups( state, siteId );
+
+	connection.preferences( locale, groups );
```

Comment:
Nice catch. I guess we could use the `withSocket` approach that we have on the happychat side. For now I just checked for `isHappychatClientConnected`